Cargo.toml

```diff
@@ -17,7 +17,7 @@
 
 [package]
 name = "datafusion-python"
-version = "29.0.0"
+version = "31.0.0"
```

Comment:
One thing I'm not sure is if this version of datafusion-python follows DataFusion version.

Jumping from 29.0.0 to 31.0.0 looks not correct to me.

cc @andygrove

Reply:
It looked to me like these previously have always matched 🤷‍♂️ But definitely happy for them not to!

Reply:
Yea, if arrow-datafusion-python always matches Rust DataFusion, it sounds good.

Reply:
We skipped 30.0.0 due to regression that broke some Python tests. See https://github.com/apache/arrow-datafusion-python/pull/475 for details.

Reply:
oops I meant to link to https://github.com/apache/arrow-datafusion-python/pull/464

Reply:
I see. Thanks @andygrove